Add backend calendar controller

damsfx opened the pull request:
Original PR #970 

**FullCalendar** [docs](https://fullcalendar.io/docs)
- [x] update to fullcalendar v6.1.15
- [x] add all available locales
- [ ] make use of package.json 

> Notes : 
> - `firstDay` can be set by fullcalendar according to his configured locale [see docs](https://fullcalendar.io/docs/firstDay) but it doesn't seems to work in our case

**CalendarController**
- Config : 
    - [x] add widget theme color for buttons (default, primary or secondary)
    - [x] add widget initial view
    - [x] add widget first day of week
    - [x] add model attribute for all day event

damsfx committed:
Merge branch 'develop' into wip/calendar-widget

damsfx committed:
Update to fullcalendar v6.1.15

-add widget initial view config
-add widget first day of week config
-add model attribute config for all day event

damsfx committed:
Clean up

LukeTowers committed:
Merge branch 'wip/calendar-widget' into wip/calendar-widget

LukeTowers commented on modules/backend/widgets/Calendar.php:
```diff
@@ -84,6 +89,16 @@ class Calendar extends WidgetBase
     public array $availableDisplayModes = [];
 
+    /**
+     * Initial calendar view,  either mont, week, day or list
```
```suggestion
     * Initial calendar view, one of month, week, day, list
```

LukeTowers committed:
Update modules/backend/widgets/Calendar.php

LukeTowers commented on modules/backend/widgets/Calendar.php:
```diff
@@ -153,7 +183,13 @@ public function init()
+            $this->recordEnd,
+        ];
+
+        // $this->validateModel();
```
Why is this commented out?

damsfx replied:
I tried to add it based on the ListController but I get an error.  
Need to take a closer look.

damsfx replied:
@LukeTowers If I try to validate the model (as in Lists widget), then this error in throw :   
`Typed property Backend\Widgets\Calendar::$model must not be accessed before initialization`.

Same if I try to add `model` to the `fillFromConfig()`method in plugin initialization.

LukeTowers commented on modules/backend/widgets/calendar/assets/less/calendar.less:
```diff
@@ -1,16 +1,30 @@
 
     .calendar-control{
-        max-width: 1000px;
+        // max-width: 1000px;
```
Commented out?

damsfx replied:
Yes, this way the widget can expend to the max width available.

LukeTowers commented on modules/backend/widgets/calendar/assets/less/calendar.less:
```diff
@@ -1,16 +1,30 @@
+
+
+// Importing file while using 
+// @import "../../../../../system/assets/ui/less/button.variables.less";
```
Commented out?

damsfx replied:
Yes, my intention was to get the brand colors of Winter, but I think the assets needs to be compile instead of directly use the less file.

damsfx replied:
The calendar css is now compiled with `winter:util compile less` command.  
The winter's button colors are used to style calendar widget buttons.

LukeTowers commented on modules/backend/widgets/calendar/assets/less/calendar.less:
```diff
@@ -1,16 +1,30 @@
+
+
```
```suggestion
```

LukeTowers committed:
Update modules/backend/widgets/calendar/assets/less/calendar.less

LukeTowers commented:
@damsfx can you take a stab at pulling and compiling the assets with vite so that they're pulled in with package.json rather than including the vendor assets in the code directly? We're trying to move away from including the raw vendor assets as the source of truth (although we'd still be committing the compiled results and including those), it's mostly just to make it easier to update the dependencies in the future. Let me know if you need any help with that.

damsfx committed:
Clean css

- remove unused files
- remove comments

damsfx commented:
I wouldn't mind a little help on this one.
I don't have much experience with Vite, I'm just starting to integrate it into my own plugins/themes.

damsfx committed:
Use compiled css

- add calendar less file to ServiceProvider
- use fullcalendar css variables for theming
- add option to chosse calendar theme for buttons style base on Winter's ones

LukeTowers commented:
Going to merge this into #970 as @jaxwilko will have some time to work on this ❤️
